assets/js/phoenix_live_view/live_socket.js

```diff
@@ -666,14 +667,55 @@ export default class LiveSocket {
       },
     );
     this.on("dragover", (e) => e.preventDefault());
+    this.on("dragenter", (e) => {
+      const dropzone = closestPhxBinding(
+        e.target,
+        this.binding(PHX_DROP_TARGET),
+      );
+
+      if (!dropzone || !(dropzone instanceof HTMLElement)) {
+        return;
+      }
+
+      if (eventContainsFiles(e)) {
+        dropzone.classList.add("phx-files-dropzone");
```

Comment:
To be consistent with what we do elsewhere, please move this into a constant in `constants.js` and import as `PHX_DROPZONE_CLASS`.

I wonder though if it should be named something like "phx-drop-target-active"?